src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -0,0 +1,467 @@
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+using namespace irr;
+using namespace asset;
+
+bool CGraphicsPipelineLoaderMTL::isALoadableFileFormat(io::IReadFile* _file) const
+{
+    if (!_file)
+        return false;
+
+    const size_t prevPos = _file->getPos();
+
+    _file->seek(0ull);
+
+    std::string mtl;
+    mtl.resize(_file->getSize());
+    _file->read(mtl.data(), _file->getSize());
+    _file->seek(prevPos);
+
+    return mtl.find("newmtl") != std::string::npos;
+}
+
+namespace
+{
+    core::smart_refctd_ptr<ICPUPipelineLayout> makePipelineLayoutFromMtl(const CMTLPipelineMetadata::SMtl& _mtl)
+    {
+        const size_t textureCnt = std::count_if(_mtl.maps, _mtl.maps + CMTLPipelineMetadata::SMtl::EMP_REFL_POSX + 1u, [](const std::string& _path) { return !_path.empty(); });
+        auto bindings = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUDescriptorSetLayout::SBinding>>(textureCnt);
+
+        ICPUDescriptorSetLayout::SBinding bnd;
+        bnd.count = 1u;
+        bnd.stageFlags = ICPUSpecializedShader::ESS_FRAGMENT;
+        bnd.type = EDT_COMBINED_IMAGE_SAMPLER;
+        std::fill(bindings->begin(), bindings->end(), bnd);
+
+        ICPUSampler::SParams smplParams;
+        smplParams.TextureWrapU = smplParams.TextureWrapV = smplParams.TextureWrapW = ICPUSampler::ETC_REPEAT;
+        smplParams.BorderColor = ICPUSampler::ETBC_FLOAT_TRANSPARENT_BLACK;
+        smplParams.MinFilter = smplParams.MaxFilter = ICPUSampler::ETF_LINEAR;
+        smplParams.MipmapMode = ICPUSampler::ESMM_LINEAR;
+        smplParams.AnisotropicFilter = 0u;
+        smplParams.CompareEnable = 0u;
+        smplParams.LodBias = 0.f;
+        smplParams.MinLod = -1000.f;
+        smplParams.MaxLod = 1000.f;
+
+        core::smart_refctd_ptr<ICPUSampler> samplers[2];
+        samplers[0] = core::make_smart_refctd_ptr<ICPUSampler>(smplParams);
+        smplParams.TextureWrapU = smplParams.TextureWrapV = smplParams.TextureWrapW = ICPUSampler::ETC_CLAMP_TO_BORDER;
+        samplers[1] = core::make_smart_refctd_ptr<ICPUSampler>(smplParams);
```

Comment:
can you add these ICPUSamplers to the asset cache under the `builtin` path, i.e. something like `irr/builtin/samplers/XXXX` when the `IAssetManager` constructor/init begins

This sort of stuff will allow us to reduce resource duplication in the future

Reply:
i see that `irr/builtin/samplers/default` has `AnisotropicFilter = 4u`. should it also be like it in those?

Reply:
well 16x Anisotropic filter (2^4) is the max on most economy-grade GPUs, so yes.

Reply:
unless you're making a `irr/builtin/samplers/nearest` XD

Reply:
@Crisspl questions?